types/preview/@ember/routing/router.d.ts

```diff
@@ -21,7 +22,7 @@ declare module '@ember/routing/router' {
      *
      * @note the `'auto'` location is [deprecated](https://deprecations.emberjs.com/v4.x/#toc_deprecate-auto-location).
      */
-    location: 'history' | 'hash' | 'none' | 'auto';
+    location: 'history' | 'hash' | 'none' | 'auto' | Location | string;
```

Comment:
The string literals and `| string` are redundant, so let's do this:

```suggestion
    location: Location | string;
```

Also, we should make sure to update the types on the implementation as well.